### ISSUE TITLE
Images drawn with transparency don't wrap

### ISSUE THREAD

**Issue by iangilman** [openseadragon/openseadragon#988]
I don't know if this is a scenario anyone cares about, but I thought I'd mention it.

There are a number of scenarios that cause us to render to an offscreen "sketch canvas" and then copy to the onscreen canvas. For instance if your image is drawn at less than 100% opacity, or if it's a png (which presumably has transparency). In these cases, if we have wrapHorizontal and/or wrapVertical turned on, those images don't wrap.

**Comment by iangilman** [comment]
Here's some repro code, for use in the /test/demo/basic.html file:

        var viewer = OpenSeadragon({
            id: "contentDiv",
            prefixUrl: "../../build/openseadragon/images/",
            tileSources: [
                {
                    width: 100,
                    tileSource: "../data/testpattern.dzi",
                    opacity: 0.2
                },
                {
                    width: 100,
                    tileSource: {
                        type: "image",
                        url: "/test/data/A.png"
                    }
                }
            ],
            wrapHorizontal: true,
            wrapVertical: true
        });

The bottom image won't wrap because of its opacity setting and the top one won't wrap because it's a png on top of another image (pngs on the base layer don't use the sketch canvas).

**Comment by VoidVolker** [comment]
Here we have three bugs: 
1) Image with opacity not wrapping - looks like this bug not in `updateLevel`, but in some draw function;
2) Multi image wrapping: update level is calling for _each_ image - as reasult we have overlapping of images;
3) After some zoom one image can replace other image - reason is bug 2.

So, looks like it is old bugs and need to add more logic to `updateLevel` for multi image case. Also, what if we want wrap one image, but not wrap other image? Maybe add additional wrap options to tilesource and handle them?

**Comment by VoidVolker** [comment]
Bug 4:
![image](https://cloud.githubusercontent.com/assets/5086438/17260454/1fc69e32-55d9-11e6-9a20-44c4b006e904.png)

Looks like it is bug of renderer.

**Comment by avandecreme** [comment]
Another issue:
When outside of the [0, 0, 1, aspectRatio] rectangle, if a high res tile is not yet available, it does not use a low res tile to cover it up.

This is for example noticeable on the Toronto panorama here: http://alexbartl.de/panorama.html
Make a 360 rotation by dragging left or right and then zoom-in. Screen goes black while high res tiles are loading.

**Comment by VoidVolker** [comment]
Yes, we know about this bug. @iangilman I'll make issue for each of this bugs.

**Comment by iangilman** [comment]
@VoidVolker Thank you for filing issues for these bugs!

**Comment by iangilman** [comment]
Note that this works fine: 

        var viewer = OpenSeadragon({
            id: "contentDiv",
            prefixUrl: "../../build/openseadragon/images/",
            tileSources: ["../data/testpattern.dzi","../data/wide.dzi"],
            wrapHorizontal: true
        });

... because neither image uses transparency.
